### PR TITLE
Change the custom error to Ballerina error

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
Change the custom error to Ballerina error

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: amazons3/amazons3_connector.bal [hunk 1]
```diff
@@ -267,3 +238,10 @@ function getClientEndpoint(string bucketName) returns http:Client {
     clientEndpoint.init(clientConfig);
     return clientEndpoint;
 }
+
+function setResponseError(int statusCode, xml xmlResponse) returns error {
```
Comment (Biruntha) [inline amazons3/amazons3_connector.bal]: Move this method to the file where we write other utils methods

Comment (kalaiyarasiganeshalingam) [reply amazons3/amazons3_connector.bal]: Fixed.

File: README.md [hunk 2]
```diff
@@ -63,7 +63,7 @@ function main(string... args) {
             io:println("Bucket Status: " + status);
         }
         //Unsuccessful attempts return an AmazonS3 error.
-        amazons3:AmazonS3Error e => io:println(e);
+        amazons3:error e => io:println(e);
```
Comment (VijithaEkanayake) [inline README.md]: This should simply be the error right? Please check all the relevant places
error e => io:println(e);

File: amazons3/amazons3_connector.bal [hunk 3]
```diff
@@ -146,39 +129,34 @@ function AmazonS3Connector::getObject(string bucketName, string objectName) retu
 
     var httpResponse = clientEndpoint->get(requestURI, message = request);
     match httpResponse {
-        error err => {
-            amazonS3Error.message = err.message;
-            amazonS3Error.cause = err.cause;
-            return amazonS3Error;
+        err => {
```
Comment (VijithaEkanayake) [inline amazons3/amazons3_connector.bal]:
```suggestion
        error err =>  {
```

File: amazons3/amazons3_connector.bal [hunk 4]
```diff
@@ -146,39 +129,34 @@ function AmazonS3Connector::getObject(string bucketName, string objectName) retu
...
         }
         http:Response response => {
             int statusCode = response.statusCode;
             var amazonResponse = response.getPayloadAsString();
             match amazonResponse {
-                error err => {
-                    amazonS3Error.message = err.message;
-                    amazonS3Error.cause = err.cause;
-                    return amazonS3Error;
+                err => {
```
Comment (VijithaEkanayake) [inline amazons3/amazons3_connector.bal]:
```suggestion
                error err => {
```

File: amazons3/amazons3_connector.bal [hunk 5]
```diff
@@ -103,39 +92,33 @@ function AmazonS3Connector::getAllObjects(string bucketName) returns S3Object[]|
...
     }
 }
 
-function AmazonS3Connector::getObject(string bucketName, string objectName) returns S3Object|AmazonS3Error {
+function AmazonS3Connector::getObject(string bucketName, string objectName) returns S3Object|error {
 
     endpoint http:Client clientEndpoint = getClientEndpoint(bucketName);
 
-    AmazonS3Error amazonS3Error = {};
+    error err = {};
```
Comment (VijithaEkanayake) [inline amazons3/amazons3_connector.bal]: Do we need to create error here, we can move this to line #147 ?

File: amazons3/amazons3_types.bal [hunk 6]
```diff
@@ -28,38 +28,38 @@ public type AmazonS3Connector object {
...
-    # + return - If success, returns Status object, else returns AmazonS3Error object
-    public function createBucket(string bucketName) returns Status|AmazonS3Error;
+    # + return - If success, returns Status object, else returns error object
+    public function createBucket(string bucketName) returns Status|error;
 
     # Retrieve the existing objects in a given bucket.
     # + bucketName - The name of the bucket
-    # + return - If success, returns S3Object[] object, else returns AmazonS3Error object
-    public function getAllObjects(string bucketName) returns S3Object[]|AmazonS3Error;
+    # + return - If success, returns S3Object[] object, else returns error object
```
Comment (VijithaEkanayake) [inline amazons3/amazons3_types.bal]: Just say returns error, not the error object, check all the other places

Comment (kalaiyarasiganeshalingam) [reply amazons3/amazons3_types.bal]: Fixed.

File: amazons3/amazons3_connector.bal (hunk 5 above)
Comment (kalaiyarasiganeshalingam) [reply amazons3/amazons3_connector.bal]: Fixed

File: amazons3/amazons3_connector.bal (hunk 4 above)
Comment (kalaiyarasiganeshalingam) [reply amazons3/amazons3_connector.bal]: Fixed

File: amazons3/amazons3_connector.bal (hunk 3 above)
Comment (kalaiyarasiganeshalingam) [reply amazons3/amazons3_connector.bal]: Fixed

File: README.md (hunk 2 above)
Comment (kalaiyarasiganeshalingam) [reply README.md]: Fixed

**Pull request merged**